src/pybind/mgr/cephadm/module.py

```diff
@@ -3195,13 +3196,25 @@ def set_custom_prometheus_alerts(self, alerts_file: str) -> str:
 
     @handle_orch_error
     def set_prometheus_target(self, url: str) -> str:
+        try:
+            parsed_url = urlparse(url)
+            host = parsed_url.hostname
+            port = parsed_url.port
+            if not host:
+                return 'Invalid URL. Hostname is missing.'
+            ipaddress.ip_address(host)
+            url = f"{host}:{port}" if port else host
```

Comment:
@epuertat upsss, good catch! Thanks. I didn't see that in my last review :see_no_evil: 

@aaSharma14 you have to wrap the host before building the endpoint target for ipv6 compatibility

```
from ceph.deployment.utils import wrap_ipv6
...
host = wrap_ipv6(host)
target = f"{host}:{port}" if port else host
```
... plz, use target or endpoint and don't call that variable url because it's not :(

Reply:
`.netloc` already gives you a valid endpoint.

Reply:
@epuertat even better :+1: